src/js/utils/dom.js

```diff
@@ -29,25 +30,28 @@ export function getEl(id){
  * @return {Element}
  * @function createEl
  */
-export function createEl(tagName='div', properties={}){
+export function createEl(tagName='div', properties={}, attributes={}){
   let el = document.createElement(tagName);
 
   Object.getOwnPropertyNames(properties).forEach(function(propName){
-      let val = properties[propName];
-
-      // Not remembering why we were checking for dash
-      // but using setAttribute means you have to use getAttribute
-
-      // The check for dash checks for the aria- * attributes, like aria-label, aria-valuemin.
-      // The additional check for "role" is because the default method for adding attributes does not
-      // add the attribute "role". My guess is because it's not a valid attribute in some namespaces, although
-      // browsers handle the attribute just fine. The W3C allows for aria- * attributes to be used in pre-HTML5 docs.
-      // http://www.w3.org/TR/wai-aria-primer/#ariahtml. Using setAttribute gets around this problem.
-      if (propName.indexOf('aria-') !== -1 || propName === 'role' || propName === 'type') {
-       el.setAttribute(propName, val);
-      } else {
-       el[propName] = val;
-      }
+    let val = properties[propName];
+
+    // See #2176
+    // We originally were accepting both properties and attributes in the
+    // same object, but that doesn't work so well.
+    if (propName.indexOf('aria-') !== -1 || propName === 'role' || propName === 'type') {
+      log.warn(`Setting attributes in the second argument of createEl()
```

Comment:
does this have a bunch of extra whitespace in it? Might want to grab https://www.npmjs.com/package/tsml